latest version (#4887)
fishtown-analytics/docs.getdbt.com: website/snippets/_config-dbt-version-check.md

Starting in 2024, when you select **Keep on latest version** in dbt Cloud, dbt will ignore the `require-dbt-version` config. Refer to [Keep on latest version](/docs/dbt-versions/upgrade-dbt-version-in-cloud#keep-on-latest-version) (available as a beta) for more details. 

dbt Labs is committed to zero breaking changes for code in dbt projects, with ongoing releases to dbt Cloud and new versions of dbt Core. We also recommend these best practices:

> 

Reviewer: I wonder if instead of having these two bullets of detail here, they should live in the pages I linked to below and we can summarize here and send them to that page for details. Just a thought.

> - **If you install dbt packages** &mdash; for use in your project, whether the package is maintained by your colleagues or a member of the open source dbt community, we recommend pinning the package to a specific revision or `version` boundary. Since v1.7, dbt manages this out-of-the-box by _locking_ the version/revision of packages in development in order to guarantee predictable builds in production. To learn more, refer to [Predictable package installs](/reference/commands/deps#predictable-package-installs).

Reviewer: Should this information also live in https://docs.getdbt.com/docs/build/packages#how-do-i-add-a-package-to-my-project?

> - **If you maintain dbt packages** &mdash; whether on behalf of your colleagues or members of the open source community, we recommend writing defensive code that checks to verify that other required packages and global macros are available. For example, if your package depends on the availability of a `date_spine` macro in the global `dbt` namespace, you can write:

Reviewer: Should this info also live in the [guide](https://docs.getdbt.com/guides/building-packages?step=1)? and/or package repos eventually? cc @dbeatty10

Author: thank you! will add this snippet as a fast-follow

> Starting in 2024, when you select **Keep on latest version** in dbt Cloud, it will ignore version checking. However, if you're not using this configuration, dbt Labs recommends:

Reviewer 2: ```suggestion
Starting in 2024, when you select **Keep on latest version** in dbt Cloud, the `require-dbt-version` config is ignored.
```

- **Writing defensive code** &mdash; If you're developing dbt code that could be run in a variety of execution contexts (such as a package) and you depend on newer dbt functionality, you can add conditional logic that checks for the presence of other packages or macros.

> - **Pinning packages** &mdash; If you're maintaining a dbt project that installs third-party packages and are concerned about the potential for breaking changes in those packages, you should pin the package to a specific revision or `version` boundary. Since v1.7, this is the default dbt behavior, by _locking_ the version/revision of packages in development in order to guarantee predictable builds in production.

Reviewer 2: I think we should reorder these bullets, and make the audience clearer for each:
- **For anyone installing packages,** we recommend...
- **For package maintainers,** we recommend writing defensive code. (I should provide an actual example of what I mean here.)

Since I wrote the first draft of this, and I have the context, I can revise this tomorrow and push up changes

> To learn more, refer to [Keep on latest version](/docs/dbt-versions/upgrade-core-in-cloud##keep-on-latest-version) which is available as a beta. 

Author: in the spirit of keeping this more evergreen, removed some of the historical context

Reviewer 2: My instinct would be to mention "beta" when we first mention "Keep on latest version" higher up - what do you think?
